### ISSUE TITLE
Panic when executing webhook notification

### ISSUE THREAD

**Issue by orthecreedence** [victorcoder/dkron#260]
Hello! We're using Dkron 0.9.2 and had an crash on our dkron agent nodes:

```
Apr 18 19:59:07 dkron02 dkron[1196]: panic: Post https://hooks.slack.com/services/TXXXXXXXXX/YYYYYYYYY/ZZZZZZZZZZZZZZZZZZ: read tcp 172.31.35.190:56046->52.84.22.122:443: read: connection reset by peer
Apr 18 19:59:07 dkron02 dkron[1196]: goroutine 4543301 [running]:
Apr 18 19:59:07 dkron02 dkron[1196]: panic(0xa14140, 0xc420471380)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/usr/local/opt/go/libexec/src/runtime/panic.go:500 +0x1a1
Apr 18 19:59:07 dkron02 dkron[1196]: github.com/victorcoder/dkron/dkron.(*Notifier).callExecutionWebhook(0xc420367690)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/notifier.go:132 +0x4ef
Apr 18 19:59:07 dkron02 dkron[1196]: github.com/victorcoder/dkron/dkron.(*Notifier).Send(0xc420367690)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/notifier.go:38 +0x7c
Apr 18 19:59:07 dkron02 dkron[1196]: github.com/victorcoder/dkron/dkron.(*RPCServer).ExecutionDone(0xc42001e9e8, 0xc42027d6a0, 0x11, 0xed088c974, 0xf4e9615, 0xdeea20, 0xed088c977, 0x2940960f, 0xdeea20, 0x1, ...)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/Users/victorcoder/src/github.com/victorcoder/dkron/dkron/rpc.go:114 +0x939
Apr 18 19:59:07 dkron02 dkron[1196]: reflect.Value.call(0xc4202bba40, 0xc42001ea00, 0x13, 0xaca299, 0x4, 0xc420367ee0, 0x3, 0x3, 0xdee840, 0xc400000001, ...)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/usr/local/opt/go/libexec/src/reflect/value.go:434 +0x5c8
Apr 18 19:59:07 dkron02 dkron[1196]: reflect.Value.Call(0xc4202bba40, 0xc42001ea00, 0x13, 0xc420367ee0, 0x3, 0x3, 0xe, 0xffffffffffffffff, 0xf)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/usr/local/opt/go/libexec/src/reflect/value.go:302 +0xa4
Apr 18 19:59:07 dkron02 dkron[1196]: net/rpc.(*service).call(0xc4202c5500, 0xc4200436c0, 0xc420283fd0, 0xc4202acf80, 0xc42027d420, 0xa87440, 0xc420528a80, 0x199, 0x995fe0, 0xc420376ab0, ...)
Apr 18 19:59:07 dkron02 dkron[1196]: #011/usr/local/opt/go/libexec/src/net/rpc/server.go:383 +0x148
Apr 18 19:59:07 dkron02 dkron[1196]: created by net/rpc.(*Server).ServeCodec
Apr 18 19:59:07 dkron02 dkron[1196]: #011/usr/local/opt/go/libexec/src/net/rpc/server.go:477 +0x421
```

Not sure if this is possible, but seemed to take down all three of the dkron agents in our cluster. I looked at the 0.9.3 changelog and it didn't mention anything like this, so thought it might still be an issue. 

Thank you for the great piece of infrastructure.

**Comment by victorcoder** [comment]
Thanks for reporting, there should be a code path that makes the app panic, this should be solved but definitely notifications should be moved to plugins.

I'll take a look

**Comment by victorcoder** [comment]
This was solved in 7d9001441fc3e152ac77f526acb2e0bd79081bac
